test/bcreader_tests.jl

```diff
@@ -0,0 +1,308 @@
+            "",                                 # varname
+            dummy_dates,                        # all_dates
+            deepcopy(segment_idx0),             # segment_idx
+            segment_idx0,                       # segment_idx0
+            monthly_fields,                     # monthly_fields
+            segment_length,                     # segment_length
+            nothing,                            # scaling_function
+            interpolate_daily,                  # interpolate_daily
+            nothing,                            # land_mask
+        )
+        @test BCReader.interpolate_midmonth_to_daily(FT, date0, bcf_info_interp) == ones(boundary_space_t) .* FT(0.5)
+
+        # test interpolate_midmonth_to_daily without interpolation
+        interpolate_daily = false
+
+        bcf_info_no_interp = BCReader.BCFileInfo(
+            FT,                                 # FT
+            "",                                 # bcfile_dir
+            ClimaComms.SingletonCommsContext(), # comms_ctx
+            "",                                 # hd_outfile_root
+            "",                                 # varname
+            dummy_dates,                        # all_dates
+            deepcopy(segment_idx0),             # segment_idx
+            segment_idx0,                       # segment_idx0
+            monthly_fields,                     # monthly_fields
+            segment_length,                     # segment_length
+            nothing,                            # scaling_function
+            interpolate_daily,                  # interpolate_daily
+            nothing,                            # land_mask
+        )
+        @test BCReader.interpolate_midmonth_to_daily(FT, date0, bcf_info_no_interp) == monthly_fields[1]
+    end
+
+    # Add tests which use TempestRemap here -
+    # TempestRemap is not built on Windows because of NetCDF support limitations
+    # `bcf_info_init` uses TR via a call to `hdwrite_regridfile_rll_to_cgll`
+    if !Sys.iswindows()
+        @testset "test update_midmonth_data! for FT=$FT" begin
+            # setup for test
+            date0 = date1 = DateTime(1979, 01, 01, 01, 00, 00)
+            date = DateTime(1979, 01, 01, 00, 00, 00)
+            tspan = (1, 90 * 86400) # Jan-Mar
+            Δt = 1 * 3600
+
+            radius = FT(6731e3)
+            Nq = 4
+            domain = Domains.SphereDomain(radius)
+            mesh = Meshes.EquiangularCubedSphere(domain, 4)
+            topology = Topologies.Topology2D(mesh)
+            quad = Spaces.Quadratures.GLL{Nq}()
+            boundary_space_t = Spaces.SpectralElementSpace2D(topology, quad)
+            current_fields = Fields.zeros(FT, boundary_space_t), Fields.zeros(FT, boundary_space_t)
+
+            land_mask_t = Fields.zeros(boundary_space_t)
+            dummy_data = (; test_data = zeros(axes(land_mask_t)))
+
+            datafile_rll = sst_data
+            varname = "SST"
```

Comment:
is this using the original file with all of the SST dates? This would make the test quite slow. I wonder if we use a more lightweight version (e.g. save only a couple of the dates in a separate file in the Box as [here](https://github.com/CliMA/ClimaCoupler.jl/wiki/ClimaCoupler-Lessons-Learned#Working-with-external-data-files)).

Reply:
This is a great idea. Do you know how we can select a subset of the dates in the file when or after we download it?